### PR TITLE
Extend `get_named()` lookup types: add `[i]startswith`

### PR DESCRIPTION
**Pull request by arkid15r**
## Proposed change

Add istartswith/startswith lookups to `get_named()`

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good (`make pre-commit`)
- [x] I've added tests to verify that the new code works and all tests pass locally (`make test`, `make tox`)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/holiday_base.py [hunk 1]
```diff
@@ -578,13 +580,36 @@ def get_named(self, holiday_name: str, lookup="icontains") -> List[date]:
...
                 if any((holiday_name_lower == name.lower() for name in names))
             ]
+        elif lookup == "istartswith":
+            holiday_name_lower = holiday_name.lower()
+            return [
+                dt
+                for dt, names in holiday_date_names_mapping.items()
+                if any(
+                    (
+                        holiday_name_lower == name[: len(holiday_name)].lower()
```
Comment (KJhellico) [inline holidays/holiday_base.py:608]:
```suggestion
                        name.lower().startswith(holiday_name_lower)
```

**Review by KJhellico — COMMENTED** [review]
I think, it will be faster

File: holidays/holiday_base.py [hunk 2]
```diff
@@ -578,13 +580,36 @@ def get_named(self, holiday_name: str, lookup="icontains") -> List[date]:
                 for dt, names in holiday_date_names_mapping.items()
                 if any((holiday_name in name for name in names))
             ]
+        elif lookup == "startswith":
+            return [
+                dt
+                for dt, names in holiday_date_names_mapping.items()
+                if any(
+                    (
+                        holiday_name == name[: len(holiday_name)]
```
[the remark this answers is not in the document]
Comment (arkid15r) [reply holidays/holiday_base.py:589]: Picking performance over readability here

File: holidays/holiday_base.py (hunk 1 above)
Comment (arkid15r) [reply holidays/holiday_base.py:608]: Ditto

**Review by arkid15r — COMMENTED** [review]
It performs slightly better with slicing

**Comment by KJhellico** [comment]
LGTM 👍

**Pull request merged**